File: src/main/java/me/zimy/questionnaire/repositories/QuestionRepository.java
package me.zimy.questionnaire.repositories;

import me.zimy.questionnaire.domain.Gender;
import me.zimy.questionnaire.domain.Question;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * This is for usage with Spring Data JPA project
 *
 * @author Dmitriy &lt;Zimy&gt; Yakovlev
 * @since 12/1/14.
 */
@Repository
public interface QuestionRepository extends MongoRepository<Question, Long> {
    List<Question> getByTargetGender(Gender gender);

    Question getByQuestion(String question);
}

File: src/main/kotlin/me/zimy/questionnaire/repositories/QuestionRepository.kt
package me.zimy.questionnaire.repositories

import me.zimy.questionnaire.domain.Gender
import me.zimy.questionnaire.domain.Question
import org.springframework.cache.annotation.Cacheable
import org.springframework.data.mongodb.repository.MongoRepository

/**
 * Created by zimy on 2/18/16.
 */
interface QuestionRepository : MongoRepository<Question, String> {
//    fun save(question: Question): Question
//    fun findOne(id: String): Question
    @Cacheable fun getByTargetGender(gender: Gender): List<Question>
    fun getByQuestion(question: String): Question
}